crates/shared/src/bad_token/token_owner_finder/solvers/solver_api.rs

```diff
@@ -16,7 +16,7 @@ pub struct SolverConfiguration {
 impl SolverConfiguration {
     /// Return type is `Token, Option<Owner>` because there are
     /// entries containing `Null` instead of owner address.
-    async fn query(&self) -> Result<HashMap<Token, Option<Owner>>> {
+    async fn query(&self) -> Result<HashMap<Token, Option<Vec<Owner>>>> {
```

Comment:
This function expects a JSON format. I believe that means the only additional "feature" `Option<Vec<Owner>>` has over `Vec<Owner>` is that this would parse:
```json
{"0xeee...eee": null}
```
If we don't care about that case then we wouldn't even have to convert the response in `get_token_owner_pairs()` if we expect `Vec<Owner>` in the response.

Reply:
That was the case before too. The result is already a Map so why would you add a key and then a null value. From an API point of view it doesn't make sense to store Option in the Map here. Dunno if we have properly documented this API anywhere or or what solvers do.

Reply:
Not sure if I'm misunderstanding you here but I totally agree that supporting the `null` case with `Option<T>` is a bit weird and simply expecting `Vec<Owner>` makes things easier for us. Maybe I expressed myself wrong.

Reply:
An empty `Vec` is a reasonable default value for `null` JSON IMO (unless we need to have this distinction somewhere, which I don't think we do).

Reply:
I was trying to say:
- You are right that Option<Vec<Owner>> should be Vec<Owner>.
- Before this PR already, Option<Owner> should have been plain Owner.
- There is no reason for this API to ever associate null with a key rather than not having the key in the first place.

Reply:
Seasolver removed the `null` so now it's safe to remove Option. Done.